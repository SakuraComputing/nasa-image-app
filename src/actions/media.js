import { SET_MEDIA, GET_MEDIA } from '../actions/types';
import axios from 'axios';

export const setMedia = (media) => ({
    type: SET_MEDIA,
    media
});

export const getMedia = (searchString, media_type) => dispatch => {

    let searchUri;

    if(media_type) {
        searchUri = `https://images-api.nasa.gov/search?q=${searchString}&media_type=${media_type}`;
    } else {
        searchUri = `https://images-api.nasa.gov/search?q=${searchString}`;
    }
    axios.get(searchUri)
    .then(res => 
        dispatch({
            type: GET_MEDIA,
            payload: res.data
        })
    )
    .catch(err => console.log(err));
}
