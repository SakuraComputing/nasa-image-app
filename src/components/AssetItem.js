import React, { Component } from 'react'

class AssetItem extends Component {
  render() {
    return (
      <div>
          <h1>Asset Item</h1>
      </div>
    )
  }
}
export default AssetItem;
