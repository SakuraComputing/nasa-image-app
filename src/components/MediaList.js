import React from 'react'

const MediaList = ({ title, item }) => {
  return (
    <div className="media-box image-frame">
      <h3>{title}</h3>
      {item.links 
        ? <img className="image-item" src={item.links[0].href} alt="noimage"/>
        : <p>{item.data[0].description}</p>}
    </div>
  )
}
export default MediaList;


