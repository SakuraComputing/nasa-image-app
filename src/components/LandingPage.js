import React, { Component } from 'react'
import { connect } from 'react-redux';
import { getMedia } from '../actions/media';
import MediaList from './MediaList';

export class LandingPage extends Component {

    constructor(props) {
        super(props);
        this.state = {
            media: [],
            searchString: '',
            imageChecked: false,
            audioChecked: false,
            videoChecked: false
        }
    }

    onChange = (e) => {
        this.setState({
            searchString: e.target.value
        })
    }

    onSearch = (e) => {

        const { searchString, imageChecked, audioChecked, videoChecked } = this.state;
        
        if(searchString) {
            if((imageChecked && audioChecked && videoChecked) || (!imageChecked && !audioChecked && !videoChecked)) {
                console.log('All checked');
                this.props.getMedia(searchString, null);
            } else if(imageChecked) {
                this.props.getMedia(searchString, 'image');
            } else if(audioChecked) {
                this.props.getMedia(searchString, 'audio');
            } else if(videoChecked) {
                console.log('Video checked')
                this.props.getMedia(searchString, 'video');
            }
        }
    }

    toggleImage = () => {
        this.setState({
            imageChecked: !this.state.checked
        })
    } 

    toggleAudio = () => {
        this.setState({
            audioChecked: !this.state.audioChecked
        })
    }
    
    toggleVideo = () => {
        this.setState({
            videoChecked: !this.state.videoChecked
        })
    }
    
    render() {

        let mediaResults;

        if(this.props.media.media) {
            const { items } = this.props.media.media.collection;
            mediaResults = items.map((item, index) => 
            (   
                <MediaList key={index} title={item.data[0].title} item={item} />
            ))
            console.log(this.state);
            console.log(this.props);
        } 

        return (
            <div className="filter-bar">
                <div className="filter-header">
                    <h1>NASA Search</h1>     
                    <div className="search-bar">
                        <input 
                            onChange={this.onChange} 
                            type="text" 
                            name="search"
                            placeholder="Enter search text"
                            className="input-box"
                        />
                        <button 
                            onClick={this.onSearch}
                            className="search-button"
                            id="test"
                        >
                        
                        <ion-icon icon="search" size="large" color="black"/>
                        </button>
                    </div> 
                    <div className="select-group">
                        <input 
                            type="checkbox" 
                            id="images" 
                            value="images" 
                            className="select-checkbox" 
                            onChange={this.toggleImage}
                        />
                        <label htmlFor="images" className="">Images</label>
                        <input 
                            type="checkbox" 
                            id="audio" 
                            value="audio" 
                            className="select-checkbox" 
                            onChange={this.toggleAudio}
                        />
                        <label htmlFor="audio" className="">Audio</label>
                        <input 
                            type="checkbox" 
                            id="video" 
                            value="video" 
                            className="select-checkbox" 
                            onChange={this.toggleVideo}
                        />
                        <label htmlFor="video" className="">Video</label>
                    </div>
                    <div className="image-container">
                        {mediaResults}
                    </div>
                </div>
            </div>
        )
    }
}
const mapStateToProps = state => ({
    media: state.media
})
export default connect(mapStateToProps, { getMedia })(LandingPage);
