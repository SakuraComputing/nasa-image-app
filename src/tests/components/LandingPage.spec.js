import React from 'react';
import { shallow } from 'enzyme';
import { LandingPage } from '../../components/LandingPage';
import media from '../fixtures/media';

describe('LandingPage', () => {

    let landingPage;
    
    beforeEach(() => {
        const getMedia = jest.fn();
        const items = jest.fn();
        landingPage = shallow((<LandingPage getMedia={getMedia} media={items}/>));
    })
    it('should render correctly', () => {
        expect(landingPage).toMatchSnapshot();
    });
    it('should set the initial `state`', () => {
        expect(landingPage.state()).toEqual({
            media: [],
            searchString: '',
            imageChecked: false,
            audioChecked: false,
            videoChecked: false
        })
    });
    describe('when typing into the input box', () => {
        const searchString = 'challenger';
        it('should accept input', () => {
            landingPage.find('.input-box').simulate('change',
                { target: { value: searchString }}
            )
            expect(landingPage.state().searchString).toEqual(searchString);
        });
    });
    describe('when selcecting the checkboxes', () => {
        const imageChecked = false;
        it('should set the state when the image checkbox is selected', () => {
            landingPage.find('#images').simulate('change', {
                target: { value: imageChecked } 
            })
            expect(landingPage.state().imageChecked).toBe(true);
        });
        const audioChecked = false;
        it('should set the state when the audio checkbox is selected', () => {
            landingPage.find('#audio').simulate('change', {
                target: { value: audioChecked } 
            })
            expect(landingPage.state().audioChecked).toBe(true);
        });
    });
});