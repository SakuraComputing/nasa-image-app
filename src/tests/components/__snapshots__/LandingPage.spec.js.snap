// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage
    getMedia={[MockFunction]}
    media={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="filter-header"
      >
        <h1>
          NASA Search
        </h1>
        <div
          className="search-bar"
        >
          <input
            className="input-box"
            name="search"
            onChange={[Function]}
            placeholder="Enter search text"
            type="text"
          />
          <button
            className="search-button"
            id="test"
            onClick={[Function]}
          >
            <ion-icon
              color="black"
              icon="search"
              size="large"
            />
          </button>
        </div>
        <div
          className="select-group"
        >
          <input
            className="select-checkbox"
            id="images"
            onChange={[Function]}
            type="checkbox"
            value="images"
          />
          <label
            className=""
            htmlFor="images"
          >
            Images
          </label>
          <input
            className="select-checkbox"
            id="audio"
            onChange={[Function]}
            type="checkbox"
            value="audio"
          />
          <label
            className=""
            htmlFor="audio"
          >
            Audio
          </label>
        </div>
        <div
          className="image-container"
        />
      </div>,
      "className": "filter-bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            NASA Search
          </h1>,
          <div
            className="search-bar"
          >
            <input
              className="input-box"
              name="search"
              onChange={[Function]}
              placeholder="Enter search text"
              type="text"
            />
            <button
              className="search-button"
              id="test"
              onClick={[Function]}
            >
              <ion-icon
                color="black"
                icon="search"
                size="large"
              />
            </button>
          </div>,
          <div
            className="select-group"
          >
            <input
              className="select-checkbox"
              id="images"
              onChange={[Function]}
              type="checkbox"
              value="images"
            />
            <label
              className=""
              htmlFor="images"
            >
              Images
            </label>
            <input
              className="select-checkbox"
              id="audio"
              onChange={[Function]}
              type="checkbox"
              value="audio"
            />
            <label
              className=""
              htmlFor="audio"
            >
              Audio
            </label>
          </div>,
          <div
            className="image-container"
          />,
        ],
        "className": "filter-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "NASA Search",
          },
          "ref": null,
          "rendered": "NASA Search",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="input-box"
                name="search"
                onChange={[Function]}
                placeholder="Enter search text"
                type="text"
              />,
              <button
                className="search-button"
                id="test"
                onClick={[Function]}
              >
                <ion-icon
                  color="black"
                  icon="search"
                  size="large"
                />
              </button>,
            ],
            "className": "search-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-box",
                "name": "search",
                "onChange": [Function],
                "placeholder": "Enter search text",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ion-icon
                  color="black"
                  icon="search"
                  size="large"
                />,
                "className": "search-button",
                "id": "test",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "color": "black",
                  "icon": "search",
                  "size": "large",
                },
                "ref": null,
                "rendered": null,
                "type": "ion-icon",
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="select-checkbox"
                id="images"
                onChange={[Function]}
                type="checkbox"
                value="images"
              />,
              <label
                className=""
                htmlFor="images"
              >
                Images
              </label>,
              <input
                className="select-checkbox"
                id="audio"
                onChange={[Function]}
                type="checkbox"
                value="audio"
              />,
              <label
                className=""
                htmlFor="audio"
              >
                Audio
              </label>,
            ],
            "className": "select-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "select-checkbox",
                "id": "images",
                "onChange": [Function],
                "type": "checkbox",
                "value": "images",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Images",
                "className": "",
                "htmlFor": "images",
              },
              "ref": null,
              "rendered": "Images",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "select-checkbox",
                "id": "audio",
                "onChange": [Function],
                "type": "checkbox",
                "value": "audio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Audio",
                "className": "",
                "htmlFor": "audio",
              },
              "ref": null,
              "rendered": "Audio",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "image-container",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="filter-header"
        >
          <h1>
            NASA Search
          </h1>
          <div
            className="search-bar"
          >
            <input
              className="input-box"
              name="search"
              onChange={[Function]}
              placeholder="Enter search text"
              type="text"
            />
            <button
              className="search-button"
              id="test"
              onClick={[Function]}
            >
              <ion-icon
                color="black"
                icon="search"
                size="large"
              />
            </button>
          </div>
          <div
            className="select-group"
          >
            <input
              className="select-checkbox"
              id="images"
              onChange={[Function]}
              type="checkbox"
              value="images"
            />
            <label
              className=""
              htmlFor="images"
            >
              Images
            </label>
            <input
              className="select-checkbox"
              id="audio"
              onChange={[Function]}
              type="checkbox"
              value="audio"
            />
            <label
              className=""
              htmlFor="audio"
            >
              Audio
            </label>
          </div>
          <div
            className="image-container"
          />
        </div>,
        "className": "filter-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              NASA Search
            </h1>,
            <div
              className="search-bar"
            >
              <input
                className="input-box"
                name="search"
                onChange={[Function]}
                placeholder="Enter search text"
                type="text"
              />
              <button
                className="search-button"
                id="test"
                onClick={[Function]}
              >
                <ion-icon
                  color="black"
                  icon="search"
                  size="large"
                />
              </button>
            </div>,
            <div
              className="select-group"
            >
              <input
                className="select-checkbox"
                id="images"
                onChange={[Function]}
                type="checkbox"
                value="images"
              />
              <label
                className=""
                htmlFor="images"
              >
                Images
              </label>
              <input
                className="select-checkbox"
                id="audio"
                onChange={[Function]}
                type="checkbox"
                value="audio"
              />
              <label
                className=""
                htmlFor="audio"
              >
                Audio
              </label>
            </div>,
            <div
              className="image-container"
            />,
          ],
          "className": "filter-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "NASA Search",
            },
            "ref": null,
            "rendered": "NASA Search",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="input-box"
                  name="search"
                  onChange={[Function]}
                  placeholder="Enter search text"
                  type="text"
                />,
                <button
                  className="search-button"
                  id="test"
                  onClick={[Function]}
                >
                  <ion-icon
                    color="black"
                    icon="search"
                    size="large"
                  />
                </button>,
              ],
              "className": "search-bar",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-box",
                  "name": "search",
                  "onChange": [Function],
                  "placeholder": "Enter search text",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <ion-icon
                    color="black"
                    icon="search"
                    size="large"
                  />,
                  "className": "search-button",
                  "id": "test",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "color": "black",
                    "icon": "search",
                    "size": "large",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "ion-icon",
                },
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="select-checkbox"
                  id="images"
                  onChange={[Function]}
                  type="checkbox"
                  value="images"
                />,
                <label
                  className=""
                  htmlFor="images"
                >
                  Images
                </label>,
                <input
                  className="select-checkbox"
                  id="audio"
                  onChange={[Function]}
                  type="checkbox"
                  value="audio"
                />,
                <label
                  className=""
                  htmlFor="audio"
                >
                  Audio
                </label>,
              ],
              "className": "select-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "select-checkbox",
                  "id": "images",
                  "onChange": [Function],
                  "type": "checkbox",
                  "value": "images",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Images",
                  "className": "",
                  "htmlFor": "images",
                },
                "ref": null,
                "rendered": "Images",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "select-checkbox",
                  "id": "audio",
                  "onChange": [Function],
                  "type": "checkbox",
                  "value": "audio",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Audio",
                  "className": "",
                  "htmlFor": "audio",
                },
                "ref": null,
                "rendered": "Audio",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "image-container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
